server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/RemoteShardsBalancer.java

```diff
@@ -266,6 +268,17 @@ void balance() {
         }
     }
 
+    /**
+     * Intended to trigger shard relocation for the shards that are submitted
+     * for tiering to go from remote node pool to local capable node
+     * pool.
+     * yet to be implemented
+     */
+    @Override
+    void tierShards() {
+        throw new UnsupportedOperationException();
```

Comment:
This probably can't be triggered by any APIs yet since we only have h2w path implemented but some more descriptive exception is always good.

Reply:
added